dockerfile

```diff
@@ -0,0 +1,21 @@
+FROM python:3.8.6-slim-buster as base
+
+RUN pip install poetry
+WORKDIR /project
+COPY /todo_app /project/todo_app
```

Comment:
This line means that whenever any file in todo_app is changed, rebuilding the image will be done from this line onwards.

To improve rebuild times, you should COPY just the toml/lock files, run poetry install, and then COPY the app files (only in the production stage - no need to copy the files for the `developments` stage because you should run the dev container with a bind mount)